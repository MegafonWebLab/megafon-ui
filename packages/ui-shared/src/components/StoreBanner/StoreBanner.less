@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-store-banner');

@containerVerticalPadding: 30px;
@gridVerticalPadding: 32px;
@gridBorderWidth: 1px;

@{b} {
    &_theme_clearWhite {
        @{b}__grid {
            background-color: @clearWhite;
        }

        @{b}__content {
            color: @freshAsphalt;
        }
    }

    &_theme_green {
        @{b}__grid {
            background-color: @green;
        }

        @{b}__content {
            color: @clearWhite;
        }
    }

    &_theme_spbSky1 {
        @{b}__grid {
            background-color: @spbSky1;
        }

        @{b}__content {
            color: @freshAsphalt;
        }
    }

    &_mask_new-iphone {
        @{b}__device {
            background-image: url('./img/new-iphone.png');
        }

        @{b}__screen {
            top: 13px;
            width: 87%;

            @media @mobileB {
                top: 10px;
            }

            @media @mobileSM {
                top: 9px;
            }
        }
    }

    &_mask_android {
        @{b}__device {
            background-image: url('./img/android.png');
        }

        @{b}__screen {
            top: 59px;
            width: 86%;

            @media @mobileB {
                top: 48px;
            }

            @media @mobileSM {
                top: 40px;
            }
        }
    }

    &_mask_white-iphone {
        @{b}__device {
            background-image: url('./img/white-iphone.png');
        }
    }

    &_mask_black-iphone {
        @{b}__device {
            background-image: url('./img/black-iphone.png');
        }
    }

    &_mask_white-iphone,
    &_mask_black-iphone {
        @{b}__screen {
            top: 57px;
            width: 87%;

            @media @mobileB {
                top: 47px;
            }

            @media @mobileSM {
                top: 39px;
            }
        }
    }

    &__container {
        @media @desktop {
            padding-top: @containerVerticalPadding;
        }
    }

    &__grid {
        padding-top: @gridVerticalPadding;
        padding-bottom: @gridVerticalPadding;
        border: @gridBorderWidth solid @spbSky2;
        border-radius: 8px;

        @media @mobileSM {
            padding-top: 24px;
            padding-bottom: 0;
        }
    }

    &__content {
        padding-left: 32px;

        @media @mobileB {
            padding-right: 10px;
        }

        @media @mobileSM {
            padding-bottom: 32px;
            padding-left: 15px;
            padding-right: 15px;
            text-align: center;
        }
    }

    &__device {
        position: relative;
        width: 235px;
        height: 1px;
        min-height: 100%;
        margin-left: auto;
        margin-right: auto;
        background-repeat: no-repeat;
        background-size: cover;
        overflow: hidden;

        @media @desktopS {
            width: 260px;
        }

        @media @mobileB {
            width: 196px;
        }

        @media @mobileSM {
            width: 163px;
        }
    }

    &__device-wrapper {
        position: relative;
        bottom: calc(2 * (@gridVerticalPadding - @gridBorderWidth));
        height: calc(100% + 2 * @gridVerticalPadding + @containerVerticalPadding);

        @media @mobile {
            position: static;
            bottom: 0;
        }

        @media @mobileB {
            height: calc(100% + @gridVerticalPadding);
        }

        @media @mobileSM {
            height: 222px;
            padding-top: 24px;
        }
    }

    &__links {
        display: flex;
    }

    &__qr-code {
        margin-right: 16px;

        @media @mobileSM {
            display: none;
        }
    }

    &__qr-code + &__stores {
        @media @mobileBU {
            flex-direction: column;
            justify-content: space-between;
        }
    }

    &__store-link {
        width: 136px;
        height: 40px;

        @media @mobileSM {
            width: 119px;
            height: 35px;
        }
    }

    &__store-link_app-store {
        margin-right: 16px;
        margin-bottom: 10px;
    }

    &__stores {
        flex-grow: 1;
        display: flex;
        flex-wrap: wrap;

        @media @mobileSM {
            justify-content: center;
        }
    }

    &__text {
        margin-bottom: 32px;
    }

    &__title {
        margin-bottom: 16px;
    }

    &__screen {
        position: absolute;
        left: 50%;
        transform: translateX(-50%);
        width: 100%;
        height: auto;
    }
}
