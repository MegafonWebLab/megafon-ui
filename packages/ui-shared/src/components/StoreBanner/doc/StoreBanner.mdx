---
name: StoreBanner
menu: Конструктор
route: components/store-banner
---

import Example from './StoreBanner.example.mdx';
import Props from './StoreBanner.props.mdx';

# StoreBanner

<CopyToClipboardBox text="import { StoreBanner } from '@megafon/ui-shared';" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
