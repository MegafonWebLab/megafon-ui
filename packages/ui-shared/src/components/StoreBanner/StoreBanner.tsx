import * as React from 'react';
import PropTypes from 'prop-types';
import { cnCreate, Grid, GridColumn, Header, Paragraph } from '@megafon/ui-core';
import StoreButton, { Theme as StoreButtonTheme, Props as StoreButtonPropsType } from '../StoreButton/StoreButton';
import './StoreBanner.less';

export const Theme = {
    CLEAR_WHITE: 'clearWhite',
    GREEN: 'green',
    SPB_SKY_1: 'spbSky1',
} as const;

type ThemeType = typeof Theme[keyof typeof Theme];

export const DeviceMask = {
    ANDROID: 'android',
    NEW_IPHONE: 'new-iphone',
    BLACK_IPHONE: 'black-iphone',
    WHITE_IPHONE: 'white-iphone',
} as const;

type DeviceMaskType = typeof DeviceMask[keyof typeof DeviceMask];

type LinkHrefType = StoreButtonPropsType['href'];
type LinkOnClickType = StoreButtonPropsType['onClick'];

export interface IStoreBannerProps {
    /** Заголовок */
    title: string;
    /** Текст */
    text: string;
    /** Ссылка на скачивание приложения в  App Store */
    linkApple: LinkHrefType;
    /** Обработчик клика по ссылке в App Store */
    onClickApple?: LinkOnClickType;
    /** Ссылка на скачивание приложения в Google Play */
    linkGoogle: LinkHrefType;
    /** Обработчик клика по ссылке в Google Play */
    onClickGoogle?: LinkOnClickType;
    /** Ссылка на картинку с QR-кодом */
    qrCode?: string;
    /** Цветовая тема компонента */
    theme?: ThemeType;
    /** Изображение телефона */
    deviceMask: DeviceMaskType;
    /** Изображение на дисплее телефона */
    imageSrc: string;
    /** Дополнительный класс */
    className?: string;
}

const cn = cnCreate('mfui-beta-store-banner');
const StoreBanner: React.FC<IStoreBannerProps> = ({
    title,
    text,
    linkApple,
    onClickApple,
    linkGoogle,
    onClickGoogle,
    qrCode,
    imageSrc,
    theme = Theme.CLEAR_WHITE,
    deviceMask,
    className,
}) => (
    <div className={cn({ theme, mask: deviceMask }, className)}>
        <div className={cn('container')}>
            <div className={cn('grid')}>
                <Grid>
                    <GridColumn
                        all="6"
                        mobile="12"
                        rightOffsetWide="1"
                    >
                        <div className={cn('content')}>
                            <Header as="h2" className={cn('title')} color="inherit">
                                {title}
                            </Header>
                            <Paragraph className={cn('text')} hasMargin={false} color="inherit">
                                {text}
                            </Paragraph>
                            <div className={cn('links')}>
                                {qrCode && (
                                    <img src={qrCode} className={cn('qr-code')} alt="QR-код"/>
                                )}
                                <div className={cn('stores')}>
                                    <StoreButton
                                        theme={StoreButtonTheme.APP_STORE}
                                        href={linkApple}
                                        onClick={onClickApple}
                                        className={cn('store-link', { 'app-store': true })}
                                    />
                                    <StoreButton
                                        theme={StoreButtonTheme.GOOGLE_PLAY}
                                        href={linkGoogle}
                                        className={cn('store-link', { 'google-play': true })}
                                        onClick={onClickGoogle}
                                    />
                                </div>
                            </div>
                        </div>
                    </GridColumn>
                    <GridColumn all="4" desktop="6" tablet="6" mobile="12">
                        <div className={cn('device-wrapper')}>
                            <div className={cn('device')}>
                                <img
                                    src={imageSrc}
                                    className={cn('screen')}
                                    alt="Изображение приложения на экране телефона"
                                />
                            </div>
                        </div>
                    </GridColumn>
                </Grid>
            </div>
        </div>
    </div>
);

StoreBanner.propTypes = {
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    linkApple: PropTypes.string.isRequired,
    onClickApple: PropTypes.func,
    linkGoogle: PropTypes.string.isRequired,
    onClickGoogle: PropTypes.func,
    qrCode: PropTypes.string,
    theme: PropTypes.oneOf(Object.values(Theme)),
    deviceMask: PropTypes.oneOf(Object.values(DeviceMask)).isRequired,
    imageSrc: PropTypes.string.isRequired,
    className: PropTypes.string,
};

export default StoreBanner;
