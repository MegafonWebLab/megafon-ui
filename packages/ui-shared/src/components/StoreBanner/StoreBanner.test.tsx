import * as React from 'react';
import { shallow } from 'enzyme';
import StoreBanner, { IStoreBannerProps, DeviceMask, Theme } from './StoreBanner';

const props: IStoreBannerProps = {
    title: 'Title',
    text: 'Text',
    linkApple: 'App Store link',
    linkGoogle: 'Google Play link',
    deviceMask: DeviceMask.ANDROID,
    imageSrc: 'image.png',
};

describe('StoreBanner', () => {
    it('should render with required props', () => {
        const wrapper = shallow(<StoreBanner {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with iphone mask', () => {
        const currentProps = {
            ...props,
            deviceMask: DeviceMask.BLACK_IPHONE,
        };
        const wrapper = shallow(<StoreBanner {...currentProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with green theme', () => {
        const currentProps = {
            ...props,
            theme: Theme.GREEN,
        };
        const wrapper = shallow(<StoreBanner {...currentProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with QR-code', () => {
        const wrapper = shallow(<StoreBanner {...props} qrCode={'qr-code.png'} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with custom class name', () => {
        const wrapper = shallow(<StoreBanner {...props} className="custom-class-name" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should call click handler on app store link', () => {
        const onClickApple = jest.fn();
        const wrapper = shallow(<StoreBanner {...props} onClickApple={onClickApple} />);

        const link = wrapper.find('.mfui-beta-store-banner__store-link_app-store');
        link.simulate('click');

        expect(onClickApple).toHaveBeenCalled();
    });

    it('should call click handler on google play link', () => {
        const onClickGoogle = jest.fn();
        const wrapper = shallow(<StoreBanner {...props} onClickGoogle={onClickGoogle} />);

        const link = wrapper.find('.mfui-beta-store-banner__store-link_google-play');
        link.simulate('click');

        expect(onClickGoogle).toHaveBeenCalled();
    });
});
