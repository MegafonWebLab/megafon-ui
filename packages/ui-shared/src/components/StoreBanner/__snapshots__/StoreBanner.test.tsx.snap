// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreBanner should render with QR-code 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_clearWhite mfui-beta-store-banner_mask_android"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <div
      className="mfui-beta-store-banner__grid"
    >
      <Grid
        multiRow={true}
      >
        <GridColumn
          all="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          rightOffsetWide="1"
        >
          <div
            className="mfui-beta-store-banner__content"
          >
            <Header
              as="h2"
              className="mfui-beta-store-banner__title"
              color="inherit"
              hAlign="inherit"
            >
              Title
            </Header>
            <Paragraph
              className="mfui-beta-store-banner__text"
              color="inherit"
              hasMargin={false}
            >
              Text
            </Paragraph>
            <div
              className="mfui-beta-store-banner__links"
            >
              <img
                alt="QR-код"
                className="mfui-beta-store-banner__qr-code"
                src="qr-code.png"
              />
              <div
                className="mfui-beta-store-banner__stores"
              >
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                  href="App Store link"
                  theme="app-store"
                />
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_google-play"
                  href="Google Play link"
                  theme="google-play"
                />
              </div>
            </div>
          </div>
        </GridColumn>
        <GridColumn
          all="4"
          desktop="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          tablet="6"
        >
          <div
            className="mfui-beta-store-banner__device-wrapper"
          >
            <div
              className="mfui-beta-store-banner__device"
            >
              <img
                alt="Изображение приложения на экране телефона"
                className="mfui-beta-store-banner__screen"
                src="image.png"
              />
            </div>
          </div>
        </GridColumn>
      </Grid>
    </div>
  </div>
</div>
`;

exports[`StoreBanner should render with custom class name 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_clearWhite mfui-beta-store-banner_mask_android custom-class-name"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <div
      className="mfui-beta-store-banner__grid"
    >
      <Grid
        multiRow={true}
      >
        <GridColumn
          all="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          rightOffsetWide="1"
        >
          <div
            className="mfui-beta-store-banner__content"
          >
            <Header
              as="h2"
              className="mfui-beta-store-banner__title"
              color="inherit"
              hAlign="inherit"
            >
              Title
            </Header>
            <Paragraph
              className="mfui-beta-store-banner__text"
              color="inherit"
              hasMargin={false}
            >
              Text
            </Paragraph>
            <div
              className="mfui-beta-store-banner__links"
            >
              <div
                className="mfui-beta-store-banner__stores"
              >
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                  href="App Store link"
                  theme="app-store"
                />
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_google-play"
                  href="Google Play link"
                  theme="google-play"
                />
              </div>
            </div>
          </div>
        </GridColumn>
        <GridColumn
          all="4"
          desktop="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          tablet="6"
        >
          <div
            className="mfui-beta-store-banner__device-wrapper"
          >
            <div
              className="mfui-beta-store-banner__device"
            >
              <img
                alt="Изображение приложения на экране телефона"
                className="mfui-beta-store-banner__screen"
                src="image.png"
              />
            </div>
          </div>
        </GridColumn>
      </Grid>
    </div>
  </div>
</div>
`;

exports[`StoreBanner should render with green theme 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_green mfui-beta-store-banner_mask_android"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <div
      className="mfui-beta-store-banner__grid"
    >
      <Grid
        multiRow={true}
      >
        <GridColumn
          all="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          rightOffsetWide="1"
        >
          <div
            className="mfui-beta-store-banner__content"
          >
            <Header
              as="h2"
              className="mfui-beta-store-banner__title"
              color="inherit"
              hAlign="inherit"
            >
              Title
            </Header>
            <Paragraph
              className="mfui-beta-store-banner__text"
              color="inherit"
              hasMargin={false}
            >
              Text
            </Paragraph>
            <div
              className="mfui-beta-store-banner__links"
            >
              <div
                className="mfui-beta-store-banner__stores"
              >
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                  href="App Store link"
                  theme="app-store"
                />
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_google-play"
                  href="Google Play link"
                  theme="google-play"
                />
              </div>
            </div>
          </div>
        </GridColumn>
        <GridColumn
          all="4"
          desktop="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          tablet="6"
        >
          <div
            className="mfui-beta-store-banner__device-wrapper"
          >
            <div
              className="mfui-beta-store-banner__device"
            >
              <img
                alt="Изображение приложения на экране телефона"
                className="mfui-beta-store-banner__screen"
                src="image.png"
              />
            </div>
          </div>
        </GridColumn>
      </Grid>
    </div>
  </div>
</div>
`;

exports[`StoreBanner should render with iphone mask 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_clearWhite mfui-beta-store-banner_mask_black-iphone"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <div
      className="mfui-beta-store-banner__grid"
    >
      <Grid
        multiRow={true}
      >
        <GridColumn
          all="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          rightOffsetWide="1"
        >
          <div
            className="mfui-beta-store-banner__content"
          >
            <Header
              as="h2"
              className="mfui-beta-store-banner__title"
              color="inherit"
              hAlign="inherit"
            >
              Title
            </Header>
            <Paragraph
              className="mfui-beta-store-banner__text"
              color="inherit"
              hasMargin={false}
            >
              Text
            </Paragraph>
            <div
              className="mfui-beta-store-banner__links"
            >
              <div
                className="mfui-beta-store-banner__stores"
              >
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                  href="App Store link"
                  theme="app-store"
                />
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_google-play"
                  href="Google Play link"
                  theme="google-play"
                />
              </div>
            </div>
          </div>
        </GridColumn>
        <GridColumn
          all="4"
          desktop="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          tablet="6"
        >
          <div
            className="mfui-beta-store-banner__device-wrapper"
          >
            <div
              className="mfui-beta-store-banner__device"
            >
              <img
                alt="Изображение приложения на экране телефона"
                className="mfui-beta-store-banner__screen"
                src="image.png"
              />
            </div>
          </div>
        </GridColumn>
      </Grid>
    </div>
  </div>
</div>
`;

exports[`StoreBanner should render with required props 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_clearWhite mfui-beta-store-banner_mask_android"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <div
      className="mfui-beta-store-banner__grid"
    >
      <Grid
        multiRow={true}
      >
        <GridColumn
          all="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          rightOffsetWide="1"
        >
          <div
            className="mfui-beta-store-banner__content"
          >
            <Header
              as="h2"
              className="mfui-beta-store-banner__title"
              color="inherit"
              hAlign="inherit"
            >
              Title
            </Header>
            <Paragraph
              className="mfui-beta-store-banner__text"
              color="inherit"
              hasMargin={false}
            >
              Text
            </Paragraph>
            <div
              className="mfui-beta-store-banner__links"
            >
              <div
                className="mfui-beta-store-banner__stores"
              >
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                  href="App Store link"
                  theme="app-store"
                />
                <StoreButton
                  className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_google-play"
                  href="Google Play link"
                  theme="google-play"
                />
              </div>
            </div>
          </div>
        </GridColumn>
        <GridColumn
          all="4"
          desktop="6"
          flex={false}
          grow={false}
          mobile="12"
          orderAll="0"
          orderDesktop="0"
          orderMobile="0"
          orderTablet="0"
          orderWide="0"
          tablet="6"
        >
          <div
            className="mfui-beta-store-banner__device-wrapper"
          >
            <div
              className="mfui-beta-store-banner__device"
            >
              <img
                alt="Изображение приложения на экране телефона"
                className="mfui-beta-store-banner__screen"
                src="image.png"
              />
            </div>
          </div>
        </GridColumn>
      </Grid>
    </div>
  </div>
</div>
`;
